pkg/kadm/topics.go

```diff
@@ -87,10 +88,53 @@ func (rs CreateTopicResponses) On(topic string, fn func(*CreateTopicResponse) er
 	return CreateTopicResponse{}, kerr.UnknownTopicOrPartition
 }
 
+// CreateTopic issues a create topics request with the given partitions,
+// replication factor, and (optional) configs for the given topic name. Under the hood,
+// this uses the default 15s request timeout and lets Kafka choose where to
+// place partitions.
+// Version 4 of the underlying create topic request was introduced in Kafka 2.4.0 and
+// brought client support for creation defaults. If talking to clusters that support
+// this request you can use -1 for partitions and replicationFactor to use the
+// default values.
+//
+// This package includes a StringPtr function to aid in building config values.
+//
+// If the topic could not be created this function will return an error. An error may be returned
+// due to authorization failure, a failed network request, a missing controller or other issues.
+func (cl *Client) CreateTopic(
+	ctx context.Context,
+	partitions int32,
+	replicationFactor int16,
+	configs map[string]*string,
+	topic string,
+) (CreateTopicResponse, error) {
```

Comment:
The `CreateTopicResponse` struct has an error property which we already check as part of the function body and return it as the second argument (error) if it is not nil. 

Should we create a separate struct for this `CreateTopic` response to make clear that the user does not have to check the `Err` property?

Reply:
I think we can just document this function fully that it returns CreateTopicResponse.Err, so if they check the return, they do not need to check the field.